src/controller/CHIPDeviceController.cpp

```diff
@@ -1178,6 +1184,298 @@ void DeviceCommissioner::OnSessionEstablished()
+    Callback::Cancelable * failureCallback = mOnCertChainFailureCallback.Cancel();
+
+    ReturnErrorOnFailure(cluster.CertChainRequest(successCallback, failureCallback, certificateChainType));
+    ChipLogDetail(Controller, "Sent Certificate Chain request, waiting for the DAC Certificate");
+    return CHIP_NO_ERROR;
+}
+
+void DeviceCommissioner::OnCertChainFailureResponse(void * context, uint8_t status)
+{
+    ChipLogProgress(Controller, "Device failed to receive the Certificate Chain request Response: 0x%02x", status);
+    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
+    commissioner->mCertChainResponseCallback.Cancel();
+    commissioner->mOnCertChainFailureCallback.Cancel();
+    // TODO: Map error status to correct error code
+    commissioner->OnSessionEstablishmentError(CHIP_ERROR_INTERNAL);
+}
+
+void DeviceCommissioner::OnCertificateChainResponse(void * context, ByteSpan certificate)
+{
+    ChipLogProgress(Controller, "Received certificate chain from the device");
+    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
+
+    commissioner->mCertChainResponseCallback.Cancel();
+    commissioner->mOnCertChainFailureCallback.Cancel();
+
+    if (commissioner->ProcessCertificateChain(certificate) != CHIP_NO_ERROR)
+    {
+        // Handle error, and notify session failure to the commissioner application.
+        ChipLogError(Controller, "Failed to process the certificate chain request");
+        // TODO: Map error status to correct error code
+        commissioner->OnSessionEstablishmentError(CHIP_ERROR_INTERNAL);
+    }
+}
+
+CHIP_ERROR DeviceCommissioner::ProcessCertificateChain(const ByteSpan & certificate)
+{
+    Transport::AdminPairingInfo * admin = mAdmins.FindAdminWithId(mAdminId);
+    VerifyOrReturnError(admin != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    switch (mCertificateChainBeingRequested)
+    {
+    case CertificateChainType::kDAC: {
+        admin->SetDeviceDAC(certificate);
+        break;
+    }
+    case CertificateChainType::kPAI: {
+        admin->SetDevicePAI(certificate);
+        break;
+    }
+    case CertificateChainType::kUnknown:
+    default: {
+        return CHIP_ERROR_INTERNAL;
+    }
+    }
+
+    VerifyOrReturnError(mState == State::Initialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mDeviceBeingPaired < kNumMaxActiveDevices, CHIP_ERROR_INCORRECT_STATE);
+
+    Device * device = &mActiveDevices[mDeviceBeingPaired];
+
+    if (admin->AreDeviceCredentialsAvailable())
+    {
+        // Step c
+        // TODO: see if vid matches
+
+        // Step b/d
+        uint16_t paiLen;
+        const uint8_t * pai = admin->GetDevicePAI(paiLen);
+        uint16_t dacLen;
+        const uint8_t * dac = admin->GetDeviceDAC(dacLen);
+
+        ReturnErrorOnFailure(ValidateCertificateChain(paa_certificate, sizeof(paa_certificate), pai, paiLen, dac, dacLen));
```

Comment:
There must be a method to look-up PAA from a cert store, which is also managed by a delegate set at startup (i.e. configured). We already have a test PAA (see https://github.com/project-chip/connectedhomeip/pull/8254) that we can start using soon for all dev devices and related work by commissioner. Over time, commissioners will have to get access to the full list of PAAs including those in DCL, and try to see if PAI/DAC chain back to them.

For the time being, I suggest:

1. Determine if PAI is in use (it is optional, it cannot be assumed)
2. If PAI exists, use its AKID to find PAA
3. If PAI doesn't exist use DAC's AKID to find PAA
4. Have a PaaStore sketch for the PAA lookup delegate API, and it should have lookup by SKID. Then it can hardcode, in the short term, only knowing about the single test PAA that will be used, until we have more complete workflows for test DACs generation.

Doing it as the above allows going with a building-blocks approach rather than having all knowledge of every data source required directly by this class.